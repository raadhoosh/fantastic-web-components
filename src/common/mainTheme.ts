import { Theme } from './styled';
import * as Color from 'color';
function darken(hex: string) {
    return Color(hex)
        .darken(0.5)
        .hex();
}
const mainTheme: Theme = {
    color: {
        primary: '#00aba9',
        primary_dark: darken('#00aba9'),
        secondary: '#0066b1',
        secondary_dark: darken('#0066b1'),
        warning: '#f0ad4e',
        warning_dark: darken('#f0ad4e'),
        light: '#f8f9fa',
        light_dark: darken('#f8f9fa'),
        dark: '#000',
        success: '#4cae4c',
        success_dark: '#155724',
        info: '#5bc0de',
        info_dark: darken('#5bc0de'),
        danger: '#d9534f',
        danger_dark: darken('#d9534f'),
        default: '#6c757d'
    },
    font: {
        size: 16,
        family: '"Roboto"',
    },
    unit: {
        padding: .1,
        font: .1,
        margin: .1,
    },
    button: {
        margin: '5px 3px',
        padding: '.375rem .75rem',
        radius: '5%',
        fontsize: '1rem',
        color: '#fff',
        border: '5px',
        shadow: '0 0 0 0.2rem #color',
        transition: `color .15s ease-in-out,
        background-color.15s ease-in-out,
        border-color .15s ease-in-out,
        box-shadow .25s ease-in-out`
    },
    alert: {
        margin: '5px 0px',
        padding: '.75rem 1.25rem',
        fontsize: '12',
        radius: '5px',
        border: '1px solid transparent',
        color: '#fff',
    },
    input: {
        fontsize: '12',
        border: '1px solid',
    },
    textarea: {
        fontsize: '12',
        width: 'auto',
        height: '50px',
    },
    label: {
        fontsize: '12',
        borderStyle: 'solid',
        borderSize: 0,
        color: '#fff',
    },
    badge: {
        fontsize: '.756rem',
        borderStyle: 'solid',
        borderSize: '1px',
        color: '#fff',
        width: 'auto',
        height: 'auto',
        radius: '10px',
    },
    h1: {
        fontsize: '2em',
        color: '#000'
    },
    h2: {
        fontsize: '1.5em',
        color: '#000'
    },
    h3: {
        fontsize: '1.17em',
        color: '#000'
    },
    h4: {
        fontsize: '1.33em',
        color: '#000'
    },
    h5: {
        fontsize: '0.83em',
        color: '#000'
    },
    h6: {
        fontsize: '0.67em',
        color: '#000'
    },
    card: {
      width: '499px'
    },
    darken: (_color?: string) => {
        return Color(_color)
            .darken(0.03)
            .hex();
    },
    hexToRgba: (hex: string, a: number) => {
        return Color(hex).fade(a ? a : 0.5).string();
    },
};

export default mainTheme;
