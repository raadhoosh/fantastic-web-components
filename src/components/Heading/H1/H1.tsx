import styled from '../../../common/styled';
import * as React from 'react';
/**
 *  properties
 */
interface IProps {
  /** This is a primary h1 it out!. */
  primary?: boolean;
  /** This is a secondary h1 it out!. */
  secondary?: boolean;
  /** This is a success h1 it out!. */
  success?: boolean;
  /** This is a info h1 it out!. */
  info?: boolean;
  /** This is a warning h1 it out!. */
  warning?: boolean;
  /** This is a danger h1 it out!. */
  danger?: boolean;
  /** This is a light h1 it out!. */
  light?: boolean;
  /** This is a dark h1 it out!. */
  dark?: boolean;
  /** This is a text color h1 it out! */
  children?: any;
  /** Description of prop "primary". */
  backgroundColor?: string;
  color?: string;
  margin?: string;
  theme?: any;
}

const EmH1 = styled('h1')(
  (props: (IProps)) => {
    const base_h1 = {
      fontSize: props.theme.h1.fontsize,
      color: props.color ? props.color : '#000',
      backgroundColor: 'transparent'
    };
    if (props.primary) {
      base_h1.color = props.theme.color.primary;
      return base_h1;
    }
    if (props.secondary) {
      base_h1.color = props.theme.color.secondary;
      return base_h1;
    }
    if (props.success) {
      base_h1.color = props.theme.color.success;
      return base_h1;
    }
    if (props.info) {
      base_h1.color = props.theme.color.info;
      return base_h1;
    }
    if (props.danger) {
      base_h1.color = props.theme.color.danger;
      return base_h1;
    }
    if (props.warning) {
      base_h1.color = props.theme.color.warning;
      return base_h1;
    }
    if (props.light) {
      base_h1.color = props.theme.color.light;
      base_h1.color = '#000';
      return base_h1;
    }
    if (props.dark) {
      base_h1.color = props.theme.color.dark;
      return base_h1;
    }

    base_h1.color = props.color ? props.color : '#000';
    return base_h1;
  }
);

/**
 * General component description in JSDoc format. Markdown is *supported*.
 */
const H1 = (props: IProps) => {

  return (
    <EmH1 {...props}> {props.children} </EmH1>
  );
};

export default H1;
