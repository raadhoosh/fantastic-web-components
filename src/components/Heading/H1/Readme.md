H1 example:

```js
<H1>This is heading 1</H1>
<H1 primary >This is heading 1 primary</H1>
<H1 secondary >This is heading 1 secondary</H1>
<H1 success >This is heading 1 success</H1>
<H1 info >This is heading 1 info</H1>
```