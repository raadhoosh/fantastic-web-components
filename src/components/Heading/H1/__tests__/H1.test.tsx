import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

// Add the custom matchers provided by 'jest-emotion'
expect.extend(createMatchers(emotion));

import H1 from '../H1';
import mainTheme from '../../../../common/mainTheme';

test('renders with correct styles', () => {
    const h1 = renderer.create(<H1 theme={mainTheme} success >ok</H1>).toJSON();
    expect(h1).toMatchSnapshot();
});
