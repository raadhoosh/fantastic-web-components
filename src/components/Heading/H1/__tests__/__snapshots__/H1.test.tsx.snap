// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
<h1
  className="css-4rg9ba"
>
   
  ok
   
</h1>
`;
